api/controllers/FlairController.js

```diff
@@ -105,13 +105,17 @@ module.exports = {
       var similar_banned_fcs = _.flatten(await* flairs.fcs.map(Flairs.getSimilarBannedFCs));
       // Get friend codes that are identical to banned users' friend codes
       var identical_banned_fcs = _.intersection(flairs.fcs, similar_banned_fcs);
-  
+
       var friend_codes = _.union(flairs.fcs, req.user.loggedFriendCodes);
 
       var newPFlair = _.get(req, "user.flair.ptrades.flair_css_class") || "default";
       var newsvFlair = _.get(req, "user.flair.svex.flair_css_class") || "";
       newsvFlair = newsvFlair.replace(/2/, "");
       var promises = [];
+      var extraFlair = req.allParams().extraFlair;
+      if (extraFlair) {
```

Comment:
We should validate that the extra flair matches one of the legal flairs in the config file, otherwise people can put arbitrary text in their flair.